Sources/StreamChatUI/ChatMessageList/ChatMessage/Decorators/ChatUnreadMessagesCountDecorationView.swift

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class ChatUnreadMessagesCountDecorationView: ChatMessageDecorationView {
+    public var content: ChatChannel? {
+        didSet {
+            updateContentIfNeeded()
+        }
+    }
+
+    #warning("No need for components here?")
+    lazy var messagesCountDecorationView = ChatMessagesCountDecorationView()
+        .withoutAutoresizingMaskConstraints
```

Comment:
It is better to have it in the components still. Imagine the customer wants to change the background colour of the `ChatMessagesCountDecorationView` for both the Thread and Channel. If we have this core/dummy view in the `Components` they only need to customize this dummy view, and it will be reflected in both.

Reply:
Hmmm, ok!